pallets/gear/src/benchmarking/syscalls.rs

```diff
@@ -229,1085 +229,704 @@ where
     }
 }
 
-pub fn alloc_bench<T>(r: u32) -> Result<Exec<T>, &'static str>
+pub(crate) struct Benches<T>
 where
     T: Config,
     T::AccountId: Origin,
 {
-    let code = WasmModule::<T>::from(ModuleDefinition {
-        memory: Some(ImportedMemory { min_pages: 0 }),
-        imported_functions: vec![ImportedFunction {
-            module: "env",
-            name: "alloc",
-            params: vec![ValueType::I32],
-            return_type: Some(ValueType::I32),
-        }],
-        handle_body: Some(body::repeated(
-            r * API_BENCHMARK_BATCH_SIZE,
-            &[
-                // Alloc 0 pages take almost the same amount of resources as another amount.
-                Instruction::I32Const(0),
-                Instruction::Call(0),
-                Instruction::Drop,
-            ],
-        )),
-        ..Default::default()
-    });
-    let instance = Program::<T>::new(code, vec![])?;
-    prepare::<T>(
-        instance.caller.into_origin(),
-        HandleKind::Handle(ProgramId::from_origin(instance.addr)),
-        vec![],
-        0u32.into(),
-    )
+    _phantom: PhantomData<T>,
 }
 
-pub fn free_bench<T>(r: u32) -> Result<Exec<T>, &'static str>
+impl<T> Benches<T>
 where
     T: Config,
     T::AccountId: Origin,
 {
-    assert!(r <= max_pages::<T>());
-
-    use Instruction::*;
-    let mut instructions = vec![];
-    for _ in 0..API_BENCHMARK_BATCH_SIZE {
-        instructions.push(I32Const(r as i32));
-        instructions.push(Call(0));
-        instructions.push(Drop);
-        for page in 0..r {
-            instructions.push(I32Const(page as i32));
-            instructions.push(Call(1));
+    fn prepare_handle(code: WasmModule<T>, value: u32) -> Result<Exec<T>, &'static str> {
+        let instance = Program::<T>::new(code, vec![])?;
+        prepare::<T>(
+            instance.caller.into_origin(),
+            HandleKind::Handle(ProgramId::from_origin(instance.addr)),
+            vec![],
+            value.into(),
+        )
+    }
+
+    pub fn alloc(r: u32) -> Result<Exec<T>, &'static str> {
+        let code = WasmModule::<T>::from(ModuleDefinition {
+            memory: Some(ImportedMemory { min_pages: 0 }),
+            imported_functions: vec!["alloc"],
+            handle_body: Some(body::repeated(
+                r * API_BENCHMARK_BATCH_SIZE,
+                &[
+                    // Alloc 0 pages take almost the same amount of resources as another amount.
```

Comment:
why anyway not to throw through bench params this? what if this change in future?

Reply:
discussed in dm